### PR TITLE
Adding support for disabling effective-pom generation

### PR DESCRIPTION
**Pull request by lpghartz**
This enables the task to work with a parent POM stored in
a Package Management feed

Added a new boolean setting to disable calling help:effective-pom during repository authentication, which prevents a parent-POM from being in a package management feed.

https://github.com/Microsoft/azure-pipelines-tasks/issues/8896

### CODE REVIEW

File: Tasks/MavenV3/task.json [hunk 1]
```diff
@@ -17,7 +17,7 @@
     "version": {
         "Major": 3,
         "Minor": 143,
```
[the remark this answers is not in the document]
Comment (lpghartz) [reply Tasks/MavenV3/task.json]: Updated ...

**Comment by azmelanar** [comment]
Hi @lpghartz! Any progress to push it forward?

**Comment by mitchdenny** [comment]
Adding @zjrunner to see if he has any input.

**Comment by azmelanar** [comment]
@lpghartz, thanks for clarification

**Comment by zjrunner** [comment]
Apologies for leaving this in this limbo state for so long.  we have been going back and forth over supportability and lifetime of the tasks in general as we approach an auth-focused task and yaml syntax set of workitems.  At this point I do want it merged to help anyone else in this state while we focus in on the auth-first set of tasks and tools.  I'll see who can pick this up, refresh the PR, and get it through.

**Comment by zjrunner** [comment]
Moved to #10637  where I fixed conflicts, added a test, and fixed the new input to not be required (lest it fail for anyone without it set today)

**Pull request closed without merging**